src/base/scoped_sched_boost.cc

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "perfetto/ext/base/scoped_sched_boost.h"
+
+#include <algorithm>
+#include <vector>
+
+#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX) || \
+    PERFETTO_BUILDFLAG(PERFETTO_OS_ANDROID)
+#include <sched.h>         // for 'SCHED_' macros and 'sched_' functions
+#include <sys/resource.h>  // for 'setpriority', 'getpriority', 'PRIO_PROCESS'
+#endif
+
+#include "perfetto/base/thread_utils.h"  // for debug logging, todo remove
+
+#include <unistd.h>
+
+#include "perfetto/ext/base/no_destructor.h"
+#include "perfetto/ext/base/status_macros.h"
+
+namespace perfetto::base {
+
+namespace {
+constexpr pid_t kCurrentPid = 0;
+
+SchedOsConfig to_sched_os_config(const SchedPolicyAndPrio& spp) {
```

Comment:
why why why 5 lines function called only in one place.
I mentioend this few days ago, i undid it, you re-did it again.
why
please. this stuff hurts readability.